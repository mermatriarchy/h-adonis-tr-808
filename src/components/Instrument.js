import React from 'react';
import Sequence from './Sequence';

export default function Instrument() {
  return (
    <>
      <div>Instrument Name Goes Here</div>
      <Sequence />
    </>
  );
}