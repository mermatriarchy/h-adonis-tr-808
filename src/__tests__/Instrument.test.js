import React from 'react';
import renderer from 'react-test-renderer';
import Instrument from '../components/Instrument';

{/* I'm adding minimal testing to this app mostly because of time constraints,
 /* but I do think tests are really important!! I''ll add some notes to the 
 /* Readme at the end of this exercise to explain the types of tests each component
 /* should have as the app grows in complexity, but for now this just tests that the
 /* Instrument component renders correctly */}

jest.mock('../components/Beat', () => () => 'beat');
const props = {
    totalBeats: [1,2,3,4],
    instrumentName: "xylophone"
} 

it('correctly renders Instrument component', () => {  
    const InstrumentComponent = renderer.create(<Instrument {...props}/>).toJSON();
    expect(InstrumentComponent).toMatchSnapshot();
});