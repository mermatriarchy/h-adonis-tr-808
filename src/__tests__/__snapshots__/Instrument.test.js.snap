// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`correctly renders Instrument component 1`] = `
<div
  className="row"
>
  <div
    className="instrument-name"
  >
    xylophone
  </div>
  beat
  beat
  beat
  beat
</div>
`;
